File: validate-transaction/src/main/java/br/com/cardoso/controller/TransactionController.java
package br.com.cardoso.controller;

import br.com.cardoso.dto.InitialTransaction;
import br.com.cardoso.model.CompletedTransaction;
import br.com.cardoso.model.TransactionStatus;
import br.com.cardoso.service.TransactionClient;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/transaction/v1")
public class TransactionController {

    private final TransactionClient transactionClient;

    public TransactionController(TransactionClient transactionClient) {
        this.transactionClient = transactionClient;
    }

    @PostMapping
    public ResponseEntity<TransactionStatus> validateTransaction(@RequestBody InitialTransaction initialTransaction) {
        CompletedTransaction completedTransaction = transactionClient.validateTransaction(initialTransaction);
        return ResponseEntity.ok(completedTransaction.transactionStatus());
    }
}

File: transaction-recorder/src/main/kotlin/br/com/cardoso/controller/TransactionController.kt
package br.com.cardoso.controller

import br.com.cardoso.service.TransactionService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/v1/transactions")
class TransactionController(private val transactionService: TransactionService) {

    @GetMapping
    fun findAllTransactions() = ResponseEntity.ok(transactionService.findAllTransactions())

    @GetMapping("/{id}")
    fun findRevisionsById(@PathVariable("id") id: Long) = ResponseEntity.ok(transactionService.findRevisionsById(id))

    //Apenas para validação do mecanismo de auditoria
    @DeleteMapping("/{id}")
    fun deleteTransactionById(@PathVariable("id") id: Long): ResponseEntity<Void> {
        transactionService.deleteTransactionById(id)
        return ResponseEntity.noContent().build()
    }
}